src/main/io/osd.c

```diff
@@ -632,32 +646,32 @@ static void osdDrawSingleElement(uint8_t item)
                 const armingDisableFlags_e flags = getArmingDisableFlags();
                 for (int i = 0; i < NUM_ARMING_DISABLE_FLAGS; i++) {
                     if (flags & (1 << i)) {
-                        tfp_sprintf(buff, "%s", armingDisableFlagNames[i]);
+                        osdFormatMessage(buff, sizeof(buff), armingDisableFlagNames[i]);
                         break;
                     }
                 }
                 break;
             }
 
             if (enabledWarnings & OSD_WARNING_BATTERY_WARNING && batteryState == BATTERY_WARNING) {
-                tfp_sprintf(buff, "LOW BATTERY");
+                osdFormatMessage(buff, sizeof(buff), "LOW BATTERY");
                 break;
             }
 
             // Show warning if battery is not fresh
             if (enabledWarnings & OSD_WARNING_BATTERY_NOT_FULL && !ARMING_FLAG(WAS_EVER_ARMED) && (getBatteryState() == BATTERY_OK)
                   && getBatteryAverageCellVoltage() < batteryConfig()->vbatfullcellvoltage) {
-                tfp_sprintf(buff, "BATT NOT FULL");
+                osdFormatMessage(buff, sizeof(buff), "BATT NOT FULL");
                 break;
             }
 
             // Visual beeper
-            if (enabledWarnings & OSD_WARNING_VISUAL_BEEPER && showVisualBeeper) {
-                tfp_sprintf(buff, "  * * * *");
+            if (enabledWarnings & OSD_WARNING_VISUAL_BEEPER && isBeeperOn()) {
```

Comment:
`showVisualBeeper` is required to latch `isBeeperOn()`, since this is only true when the beeper is active. Removing it will make the visual beeper miss beeps.